Sources/Internal/Logger/DataConnectLogger.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import OSLog
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+class DataConnectLogger {
+  private static let logger = Logger(
+    subsystem: "com.google.firebase",
+    category: "[FirebaseDataConnect]"
+  )
+
+  private static let logPrefix = "\(Version.sdkVersion) - [FirebaseDataConnect]"
+
+  static let logLevel = FirebaseConfiguration.shared.loggerLevel()
```

Comment:
I believe this will need to be a computed property, otherwise, FDC's stored logging level won't pick up changes that customers make with the setter
```suggestion
  static var logLevel = { FirebaseConfiguration.shared.loggerLevel() } 
```

Reply:
Which means that users are able to change the FirebaseConfiguration after app.config()?

Reply:
Makes sense to do this as a computed property since its value is anyway read from the Shared configuration.